test/SerialIOTest.cpp

```diff
@@ -1087,10 +1089,407 @@ TEST_CASE( "no_serial_hdf5", "[serial][hdf5]" )
+        s.setAttribute("vecUint64", std::vector< uint64_t >({18446744073709551614u, 18446744073709551615u}));
+        s.setAttribute("vecFloat", std::vector< float >({0.f, 3.40282e+38f}));
+        s.setAttribute("vecDouble", std::vector< double >({0., 1.79769e+308}));
+        s.setAttribute("vecLongdouble", std::vector< long double >({0.L, 1.18973e+4932L}));
+        s.setAttribute("vecString", std::vector< std::string >({"vector", "of", "strings"}));
+        s.setAttribute("bool", true);
+    }
+
+    Series s = Series::read("../samples/dtype_test.bp");
+
+    REQUIRE(s.getAttribute("char").get< char >() == 'c');
+    REQUIRE(s.getAttribute("uchar").get< unsigned char >() == 'u');
+    REQUIRE(s.getAttribute("int16").get< int16_t >() == 16);
+    REQUIRE(s.getAttribute("int32").get< int32_t >() == 32);
+    REQUIRE(s.getAttribute("int64").get< int64_t >() == 64);
+    REQUIRE(s.getAttribute("uint16").get< uint16_t >() == 16u);
+    REQUIRE(s.getAttribute("uint32").get< uint32_t >() == 32u);
+    REQUIRE(s.getAttribute("uint64").get< uint64_t >() == 64u);
+    REQUIRE(s.getAttribute("float").get< float >() == 16.e10f);
+    REQUIRE(s.getAttribute("double").get< double >() == 1.e64);
+    REQUIRE(s.getAttribute("longdouble").get< long double >() == 1.e80L);
+    REQUIRE(s.getAttribute("string").get< std::string >() == "string");
+    REQUIRE(s.getAttribute("vecChar").get< std::vector< char > >() == std::vector< char >({'c', 'h', 'a', 'r'}));
+    REQUIRE(s.getAttribute("vecInt16").get< std::vector< int16_t > >() == std::vector< int16_t >({32766, 32767}));
+    REQUIRE(s.getAttribute("vecInt32").get< std::vector< int32_t > >() == std::vector< int32_t >({2147483646, 2147483647}));
+    REQUIRE(s.getAttribute("vecInt64").get< std::vector< int64_t > >() == std::vector< int64_t >({9223372036854775806, 9223372036854775807}));
+    REQUIRE(s.getAttribute("vecUchar").get< std::vector< char > >() == std::vector< char >({'u', 'c', 'h', 'a', 'r'}));
+    REQUIRE(s.getAttribute("vecUint16").get< std::vector< uint16_t > >() == std::vector< uint16_t >({65534u, 65535u}));
+    REQUIRE(s.getAttribute("vecUint32").get< std::vector< uint32_t > >() == std::vector< uint32_t >({4294967294u, 4294967295u}));
+    REQUIRE(s.getAttribute("vecUint64").get< std::vector< uint64_t > >() == std::vector< uint64_t >({18446744073709551614u, 18446744073709551615u}));
+    REQUIRE(s.getAttribute("vecFloat").get< std::vector< float > >() == std::vector< float >({0.f, 3.40282e+38f}));
+    REQUIRE(s.getAttribute("vecDouble").get< std::vector< double > >() == std::vector< double >({0., 1.79769e+308}));
+    REQUIRE(s.getAttribute("vecLongdouble").get< std::vector< long double > >() == std::vector< long double >({0.L, 1.18973e+4932L}));
+    REQUIRE(s.getAttribute("vecString").get< std::vector< std::string > >() == std::vector< std::string >({"vector", "of", "strings"}));
+    REQUIRE(s.getAttribute("bool").get< uint8_t >() == static_cast< uint8_t >(true));
```

Comment:
can you add a comment on bool reads please?
maybe we also want to write this in the manual in a section on "backends: features & limitations" or so.